backend/e4h-services/ingestion-service/app/api/endpoints/file_ingestion.py

```
@@ -1437,6 +1439,83 @@ async def upload_facility_selection_excel_sheet(
             os.unlink(input_temp_file.name)
 
 
+@router.post('/icc-reports',
+             summary='Upload an ICC report Excel file, validate it against the selected System Type, '
+                     'convert it to JSON, and store it via field-planner',
+             response_description='Returns the field-planner template creation response')
+async def upload_icc_report(
+        icc_file: UploadFile = File(..., description="ICC Report Excel file (.xlsx)"),
```

Ruff (0.15.20)

[warning] Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)